packages/frontend/src/ee/features/aiCopilot/components/AiAgentPageLayout/AiAgentPageLayout.tsx

```diff
@@ -34,113 +32,140 @@ interface Props extends PropsWithChildren {
     Header?: ReactNode;
 }
 
-interface AiAgentPageLayoutRef {
-    collapseSidebar: () => void;
-    expandSidebar: () => void;
-    toggleSidebar: () => void;
-}
+export const AiAgentPageLayout: React.FC<Props> = ({
+    Sidebar,
+    Header,
+    children,
+}) => {
+    const sidebarPanelRef = useRef<ImperativePanelHandle>(null);
+    const artifactPanelRef = useRef<ImperativePanelHandle>(null);
+    const [isSidebarCollapsed, setIsSidebarCollapsed] = useState(false);
+    const [contextArtifact, setContextArtifact] = useState<ReactNode>(null);
+
+    const updateCollapsedState = () => {
+        const isCollapsed = sidebarPanelRef.current?.isCollapsed() ?? false;
+        setIsSidebarCollapsed(isCollapsed);
+    };
+
+    useEffect(() => {
+        const timer = setInterval(updateCollapsedState, 100);
+        return () => clearInterval(timer);
+    }, []);
```

Comment:
what does this do?

Reply:
`react-resizable-panels` doesn't have a reactive way of knowing if a panel is collapsed, so this adds a timer that checks and updates the state 🫠